Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -180,22 +181,28 @@ function isNullOrEmpty(str) {
 }
 
 // Creates the string that comprises the build summary text, given the location of the /sonar build folder.
-function createSonarQubeBuildSummary(sqBuildFolder: string): string {
-    // Task report is not created for PR builds
+function createSonarQubeBuildSummary(sqBuildFolder:string):Q.Promise<string> {
```

Comment:
This sonarqube-common class is getting too crowded. Please move all the report related details to a separate file. Also, please uses classes and methods, not functions.